core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -3722,6 +3731,8 @@ object LogTest {
         record(baseOffset + 2)))
       segment.append(MemoryRecords.withRecords(baseOffset + Int.MaxValue - 1, CompressionType.NONE, 0,
         record(baseOffset + Int.MaxValue - 1)))
+      Log.offsetIndexFile(logDir, baseOffset).createNewFile()
+      Log.timeIndexFile(logDir, baseOffset).createNewFile()
```

Comment:
Do we need to explicitly create these 2 index files?

Reply:
We need. Otherwise `recoverSegment()` will be invoked because the offset index is missing and truncation will happen so that the assertion here will fail: https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/log/LogTest.scala#L3690

Btw, this happened to work without this patch because the index file will always get created before the offset index file existence check (See: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/AbstractIndex.scala#L112), which means we will never reach https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/Log.scala#L475 before this patch.

Reply:
Ok, Probably add a comment why we need to explicitly create those files.

Reply:
Done.